runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -2344,31 +2300,22 @@ static String getContainerImageForJob(DataflowPipelineOptions options) {
   }
 
   static boolean useUnifiedWorker(DataflowPipelineOptions options) {
-    return hasExperiment(options, "beam_fn_api")
-        || hasExperiment(options, "use_runner_v2")
-        || hasExperiment(options, "use_unified_worker");
+    return hasExperiment(options, "beam_fn_api");
   }
 
   static boolean useStreamingEngine(DataflowPipelineOptions options) {
     return hasExperiment(options, GcpOptions.STREAMING_ENGINE_EXPERIMENT)
         || hasExperiment(options, GcpOptions.WINDMILL_SERVICE_EXPERIMENT);
   }
 
-  static void verifyDoFnSupported(
-      DoFn<?, ?> fn, boolean streaming, boolean workerV2, boolean streamingEngine) {
+  static void verifyDoFnSupported(DoFn<?, ?> fn, boolean streaming, boolean streamingEngine) {
     if (streaming && DoFnSignatures.requiresTimeSortedInput(fn)) {
       throw new UnsupportedOperationException(
           String.format(
               "%s does not currently support @RequiresTimeSortedInput in streaming mode.",
               DataflowRunner.class.getSimpleName()));
     }
     if (DoFnSignatures.usesSetState(fn)) {
-      if (workerV2) {
```

Comment:
Why do we want to remove SetState and MapState check here and below? There is no SDK support over fnapi yet so we should fail here. When we have SDK support, we can remove this check even Dataflow doesn't support.